### ISSUE TITLE
Bottom bar implementation

### ISSUE THREAD

**Issue by plarus** [freakified/TimeStylePebble#99]
For information, I am currently working on a bottom bar implementation that can display up to four widgets.
It is still a work in progress.
I do not know if it will be possible to merge this work in the main branch when finished.
Some previews of the current state of my work:

![screen2](https://cloud.githubusercontent.com/assets/16477554/20861111/0d5d910c-b988-11e6-8c6a-52dcddfbc264.png)

![screen1](https://cloud.githubusercontent.com/assets/16477554/20861112/115d55f8-b988-11e6-8e65-ef1ac0c2e145.png)

**Comment by tilden** [comment]
Oh, this is actually pretty cool. I like your approach to the date format, where we just eliminate the date widget altogether and replace it with the text.

As for integrating this, I believe it could be done, but we'd need to implement a switch for it in the settings, and some refactors would probably be necessary to facilitate switching between the two modes.

Visually, I think it's actually a good idea, it just will need a few refinements. Here's what I'm kinda thinking:
![untitled-1](https://cloud.githubusercontent.com/assets/466332/20890334/78c167aa-bad4-11e6-8032-4eb718a52dd7.png)
This centers the time horizontally, centers the colon, and vertically aligns the text and icons with each other. In terms of implementation, I'm not sure how we could achieve that last item, but I believe it could be done.

**Comment by tilden** [comment]
So I was thinking more about how we could implement the evenly-aligned bottom bar, and this is the simplest solution I could think of:

We add a second global sidebar control variable, similar to "use compact mode" now, that causes sidebar widgets to draw with the same heights. Not sure how we can get forecast to fit, but maybe that can draw it horizontally or something.

**Comment by plarus** [comment]
It is more beautiful with your correction! I have started to implement them.
The new positioning of widgets elements will require some work.
I also have some work updating the html settings page.

After that I expect to allow users to set date fields order and part to be displayed (day of week, day number, month name, month number, year)

**Comment by tilden** [comment]
Yeah, I'm thinking that maybe maybe it would be simpler to publish as second watchface entirely, which would simplify the configuration page considerably.

Also, now that Pebble has gone out of business, the app store probably won't have long to live...

**Comment by plarus** [comment]
It may just be a revival.
If the pebble store disappear, a replacement apps store may appear ...

I will continue my work on this application.
If you do not went to merge my work (when finished), I will publish it as a new application.

**Comment by belzo13** [comment]
Any progress on the bottom bar watchface. I would love to have option of four widgets!

**Comment by plarus** [comment]
It is still a work in progress. But very usable by now. Some work are needed in order to have a better config webpage display. I can send you a build version if you want.

**Comment by belzo13** [comment]
Would love to have whatever you could offer me. Is a build version ready to install? Thanks!

**Comment by plarus** [comment]
You can access to a first pre-release of my work here: https://github.com/plarus/TimeStylePebble/releases

**Comment by belzo13** [comment]
Very nice! I like it! Thank You

**Comment by plarus** [comment]
Thank you!
If you find an issue you can report it in my own issue list (https://github.com/plarus/TimeStylePebble/issues) as this work is not currently merged in the official version.

**Comment by Leonick** [comment]
Hey Plarus,
If you would do something similar for the Round I for one would really appreciate that. Obviously not going to gain space for more complications, still, complications top and bottom with time as large as possible in middle would be really nice.

**Comment by plarus** [comment]
Hi Leonick,
Did you expect something like that?

![timestylepebble round horizontal bar](https://user-images.githubusercontent.com/16477554/31587955-224767b2-b1eb-11e7-8b29-0f0cb36950f5.png)

**Comment by Leonick** [comment]
Pretty much, yea. Just make the time larger, the default layout has a few pixels of margin between the time, the widgets, and screen edge.

The widgets are off centre, but I assume that’s just on account of it being WIP.

**Comment by plarus** [comment]
You can access to my work here: https://github.com/plarus/TimeStylePebble/releases/tag/0.8.1

**Comment by Leonick** [comment]
Been meaning to get around to this, but kept forgetting...

The changes are working great, loving the watchface. Thank you.

**Comment by plarus** [comment]
I have released a new version with various corrections and aplite compatibility.

You can download it here : https://github.com/plarus/TimeStyleBBPebble/releases/tag/0.9.0

**Comment by PeacheyMcKeitch** [comment]
Any plans to add this into TimeStyle or update TimeStyle BB now that Pebble is back (woo!) out of interest.

Timestyle BB still works great btw (its my usual go to watch face though have been plaining with TimeStyle and NineTime a bit), just curious. :)

**Comment by tilden** [comment]
@PeacheyMcKeitch I updated TimeStyle recently! It now has a UV Index widget and slightly better weather! Some other new features coming also!

**Comment by PeacheyMcKeitch** [comment]
Thanks for updating it. :) Been my main watch face  for a while now. I do miss the bottom bar variant a bit because you can stick an extra info widget at the bottom but doesn’t seem like BB is getting updates any more so switched over.
